sum/front/src/components/contractInteraction.tsx

```diff
@@ -29,33 +55,52 @@ export default function ContractInteraction() {
     const calculateSum = async () => {
         setLoading(true);
         try {
-            await account?.callSC(
-                CONTRACT_ADDRESS,
-                "sum",
-                new Args().addI64(BigInt(num1)).addI64(BigInt(num2)),
-                BigInt(1)
-            );
-
-            setResult((await getLastResult()).toString());
+            setTimeout(async () => {
```

Comment:
What is the purpose of this setTimeout

Reply:
It's to ensure that the setLoading(true); updates the UI before we make the call to the smart contract.